### PR TITLE
Revive the remote cache service testing

### PR DESCRIPTION
**Pull request by gtristan**
This revives the tests which run under `tox -- --remote-cache`, one of which runs against `bst-artifact-server` and another which runs against the corresponding buildbarn services.

This also refactors the tests a bit to improve on code reuse, and adds a new `service-tests` matrix to run the tests which require running an additional service in the background while the `bst` frontend tests run.

### CODE REVIEW

File: .github/compose/ci.buildbarn-remote-cache.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#  - docker-compose -f ci.buildbarn-remote-cache.yml down
+
+version: '3.4'
+
+services:
+  bb-asset:
+    image: buildbarn/bb-remote-asset:20200903T103837Z-90136c4
+    command: /config/asset.jsonnet
+    restart: unless-stopped
+    expose:
```
Comment (cs-shadow) [inline .github/compose/ci.buildbarn-remote-cache.yml:29]: i think all these `expose` fields are redundant given that we expose them using `ports` anyway on the line below

File: .github/compose/ci.docker-compose.yml [hunk 2]
```diff
@@ -58,6 +58,45 @@ services:
...
+    <<: *tests-template
+    command: tox -vvvvv -- --color=yes --remote-cache
+    environment:
+      TOXENV: ${CI_TOXENV_MAIN}
+      ARTIFACT_CACHE_SERVICE: http://localhost:50052
+
+    # We need to use host networking mode in order to be able to
+    # properly resolve services exposed by adjacent containers.
+    #
+    network_mode: host
```
Comment (cs-shadow) [inline .github/compose/ci.docker-compose.yml:85]: We don't have to resolve it at this time, but it'll be nice to not have to use host networking mode. Perhaps by sharing a docker network between this compose file and the ones for other services. Not blocking for now.

**Review by cs-shadow — APPROVED** [review]
a couple of minor comments, but looks good overall to me.

File: .github/compose/ci.docker-compose.yml (hunk 2 above)
Comment (gtristan) [reply .github/compose/ci.docker-compose.yml:85]: removed this now

File: .github/compose/ci.buildbarn-remote-cache.yml (hunk 1 above)
Comment (gtristan) [reply .github/compose/ci.buildbarn-remote-cache.yml]: Removed this now

File: .github/compose/ci.docker-compose.yml (hunk 2 above)
Comment (gtristan) [reply .github/compose/ci.docker-compose.yml:85]: Yeah I struggled with this, won't be fixing this now.

**Pull request merged**